RubberduckTests/Inspections/EmptyIfBlockInspectionTests.cs

```diff
@@ -0,0 +1,958 @@
+        public void EmptyIfBlock_ElseIfBlockHasExecutableStatement()
+        {
+            const string inputCode =
+@"Sub Foo()
+    If True Then
+        Dim d
+    ElseIf False Then
+        Dim b
+    End If
+End Sub";
+
+            IVBComponent component;
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out component);
+            var state = MockParser.CreateAndParse(vbe.Object);
+
+            var inspection = new EmptyIfBlockInspection(state);
+            var inspector = InspectionsHelper.GetInspector(inspection);
+            var inspectionResults = inspector.FindIssuesAsync(state, CancellationToken.None).Result;
+
+            Assert.IsFalse(inspectionResults.Any());
+        }
+
+        [TestMethod]
+        [TestCategory("Inspections")]
+        public void EmptyIfBlock_QuickFixRemovesLoneIf()
+        {
+            const string inputCode =
+@"Sub Foo()
+    If True Then
+    End If
+End Sub";
+            const string expectedCode =
+@"Sub Foo()
+    
+End Sub";
+
+            IVBComponent component;
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out component);
+            var state = MockParser.CreateAndParse(vbe.Object);
+
+            var inspection = new EmptyIfBlockInspection(state);
+            var inspector = InspectionsHelper.GetInspector(inspection);
+            var inspectionResults = inspector.FindIssuesAsync(state, CancellationToken.None).Result;
+
+            new RemoveEmptyIfBlockQuickFix(state).Fix(inspectionResults.First());
+
+            Assert.AreEqual(expectedCode, state.GetRewriter(component).GetText());
+        }
+
+        [TestMethod]
+        [TestCategory("Inspections")]
+        public void EmptyIfBlock_QuickFixRemovesSingleLineIf()
+        {
+            const string inputCode =
+@"Sub Foo()
+    If True Then Else Bar
+End Sub
+
+Sub Bar()
+End Sub";
+
+            const string expectedCode =
+@"Sub Foo()
+    If Not (True) Then Bar
+End Sub
+
+Sub Bar()
+End Sub";
+
+            IVBComponent component;
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out component);
+            var state = MockParser.CreateAndParse(vbe.Object);
+
+            var inspection = new EmptyIfBlockInspection(state);
+            var inspector = InspectionsHelper.GetInspector(inspection);
+            var inspectionResults = inspector.FindIssuesAsync(state, CancellationToken.None).Result;
+
+            new RemoveEmptyIfBlockQuickFix(state).Fix(inspectionResults.First());
+
+            Assert.AreEqual(expectedCode, state.GetRewriter(component).GetText());
+        }
+
+        [TestMethod]
+        [TestCategory("Inspections")]
+        public void EmptyIfBlock_QuickFixRemovesSingleLineIf_NoWhiteSpace()
+        {
+            const string inputCode =
+@"Sub Foo()
+    If True Then Else(Bar)
+End Sub
+
+Sub Bar()
+End Sub";
+
+            const string expectedCode =
+@"Sub Foo()
+    If Not (True) Then (Bar)
```

Comment:
It's just because we have an issue for implementing an inspection that spots redundant parens; from the user's perspective it's "well, if you know they're redundant, why did you put them there in the first place?" :wink: